docs/whats-new/dotnet-aspire-9.5.md

```diff
@@ -0,0 +1,1078 @@
+var embeddingModel = builder.AddGitHubModel("embeddings", GitHubModel.OpenAI.OpenAITextEmbedding3Large);
+```
+
+For more information, see:
+
+- [.NET Aspire GitHub Models integration (Preview)](../github/github-models-integration.md).
+- [.NET Aspire Azure AI Foundry integration (Preview)](../azureai/azureai-foundry-integration.md).
+
+### Dev Tunnels hosting integration
+
+Aspire 9.5 introduces first-class support for Dev Tunnels, enabling seamless integration of secure public tunnels for your applications during development and testing scenarios.
+
+**Features:**
+
+- **Secure public tunnels**: Create public HTTPS endpoints for applications running locally.
+- **Automatic tunnel management**: Tunnels are created, configured, and cleaned up automatically.
+- **Private and anonymous tunnels**: Support for both authenticated private tunnels and public anonymous access.
+- **Development workflow integration**: Perfect for webhook testing, mobile app development, and external service integration.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Add a basic Dev Tunnel resource (default: private access)
+var tunnel = builder.AddDevTunnel("dev-tunnel");
+
+// Add your web application
+var webApp = builder.AddProject<Projects.WebApp>("webapp");
+
+// Connect the tunnel to the web application endpoint
+tunnel.WithReference(webApp.GetEndpoint("http"));
+
+builder.Build().Run();
+```
+
+The Dev Tunnels integration automatically handles Azure authentication, tunnel lifecycle management, and provides public or private URLs (depending on configuration) to connected resources, making it easy to expose local development services securely to external consumers. Dev Tunnels also improves support for mobile dev, such as .NET MAUI, making it easy to launch both your backend and mobile app at once without complex dev-time config. For more information, see [.NET Aspire Dev Tunnels integration (Preview)](../extensibility/dev-tunnels-integration.md).
+
+### YARP static files support
+
+Aspire 9.5 adds comprehensive static file serving capabilities to the YARP integration, enabling you to serve static assets directly from YARP alongside reverse proxy functionality. This is perfect for single-page applications, frontend assets, and hybrid scenarios where you need both static content and API proxying.
+
+**Features:**
+
+- **Direct static file serving**: Serve HTML, CSS, JS, and other static assets from YARP.
+- **Flexible source options**: Bind mount local directories or use Docker multi-stage builds.
+- **Automatic configuration**: Simple API enables static files with minimal setup.
+- **Production ready**: Works in both development and publish scenarios.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Enable static file serving (serves from wwwroot folder)
+var yarp = builder.AddYarp("gateway")
+    .WithStaticFiles();
+
+builder.Build().Run();
+```
+
+**Docker multi-stage build scenario:**
```

Comment:
We should point to docs for this because its not clear what this means.

Reply:
Docker docs?

https://docs.docker.com/build/building/multi-stage/

Or our stuff, I wrote some stuff about multi-stage builds in .NET, thoughts?

https://review.learn.microsoft.com/en-us/dotnet/core/docker/build-container?branch=main&tabs=windows&pivots=dotnet-9-0#create-the-dockerfile